services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/internal/update/UpdateExpressionConverter.java

```diff
@@ -0,0 +1,188 @@
+                             Stream.concat(expression.removeActions().stream().map(RemoveUpdateAction::expressionNames),
+                                           Stream.concat(expression.deleteActions().stream()
+                                                                   .map(DeleteUpdateAction::expressionNames),
+                                                         expression.addActions().stream()
+                                                                   .map(AddUpdateAction::expressionNames))));
+    }
+
+    private static Map<String, AttributeValue> mergeExpressionValues(UpdateExpression expression) {
+        return streamOfExpressionValues(expression)
+            .reduce(Expression::joinValues)
+            .orElseGet(HashMap::new);
+    }
+
+    private static Stream<Map<String, AttributeValue>> streamOfExpressionValues(UpdateExpression expression) {
+        return Stream.concat(expression.setActions().stream().map(SetUpdateAction::expressionValues),
+                             Stream.concat(expression.deleteActions().stream().map(DeleteUpdateAction::expressionValues),
+                                           expression.addActions().stream().map(AddUpdateAction::expressionValues)));
+    }
+
+    private static Map<String, String> mergeExpressionNames(UpdateExpression expression) {
+        return streamOfExpressionNames(expression)
+            .reduce(Expression::joinNames)
+            .orElseGet(HashMap::new);
+    }
+
+    private static List<String> listPathsWithoutTokens(UpdateExpression expression) {
+        return Stream.concat(expression.setActions().stream().map(SetUpdateAction::path),
+                             Stream.concat(expression.removeActions().stream().map(RemoveUpdateAction::path),
+                                           Stream.concat(expression.deleteActions().stream().map(DeleteUpdateAction::path),
+                                                         expression.addActions().stream().map(AddUpdateAction::path))))
+                     .map(UpdateExpressionConverter::removeNestingAndListReference)
+                     .filter(attributeName -> !attributeName.contains("#"))
+                     .collect(Collectors.toList());
+    }
+
+    private static List<String> listAttributeNamesFromTokens(UpdateExpression updateExpression) {
+        return mergeExpressionNames(updateExpression).values().stream()
+                                                     .map(UpdateExpressionConverter::removeNestingAndListReference)
+                                                     .collect(Collectors.toList());
+    }
+
+    private static String removeNestingAndListReference(String attributeName) {
+        return attributeName.substring(0, getRemovalIndex(attributeName));
+    }
+
+    private static int getRemovalIndex(String attributeName) {
+        for (char c : attributeName.toCharArray()) {
+            if (c == '.' || c == '[') {
```

Comment:
We have constants declared for other literal values like these, can we do the same here?

Reply:
yes